controller/api/public/stat_summary.go

```diff
@@ -92,50 +95,65 @@ func (s *grpcServer) StatSummary(ctx context.Context, req *pb.StatSummaryRequest
 		}
 	}
 
-	statTables := make([]*pb.StatTable, 0)
+	kind := req.GetSelector().GetResource().GetType()
+	if !isSupportedKind(kind) {
+		return nil, util.GRPCError(status.Errorf(codes.Unimplemented, "unimplemented resource type: %s", kind))
+	}
 
-	var resourcesToQuery []string
-	if req.Selector.Resource.Type == k8s.All {
-		resourcesToQuery = k8s.StatAllResourceTypes
-	} else {
-		resourcesToQuery = []string{req.Selector.Resource.Type}
+	// get stats for individual workload kind
+	if isK8sWorkloadKind(kind) {
```

Comment:
Maybe there's some DRYness to be gained here if we reuse `getStatsForAllKinds()` instead of `s.k8sResourceQuery()`? The former could receive the list of kinds to query, which would be just `kind` in this case, and `k8s.StatAllWorkloadKinds` in `kind == k8s.All`'s case...